pip/cmdoptions.py

```diff
@@ -354,9 +363,53 @@ def editable():
     action='store_false',
     default=True,
     help=('Do not Find and prefer wheel archives when searching indexes and '
-          'find-links locations.'),
+          'find-links locations. DEPRECATED in favour of --no-binary.'),
 )
 
+
+def _get_format_control(values, option):
+    """Get a format_control object."""
+    return getattr(values, option.dest)
+
+
+def _handle_no_binary(option, opt_str, value, parser):
+    existing = getattr(parser.values, option.dest)
+    fmt_ctl_handle_mutual_exclude(
+        value, existing.no_binary, existing.only_binary)
+
+
+def _handle_only_binary(option, opt_str, value, parser):
+    existing = getattr(parser.values, option.dest)
+    fmt_ctl_handle_mutual_exclude(
+        value, existing.only_binary, existing.no_binary)
+
+
+def no_binary():
+    return Option(
+        "--no-binary", dest="format_control", action="callback",
```

Comment:
Hmm, so there is a problem about "Binary" when it means compiled and "Binary" when it means wheels in general. One problem I'd like to address is baking the format names directly into the UI of pip. I was thinking about sdist 2.0 and what we'd end up calling that and an idea that popped into my head was "Source Wheel" (similar to RPM and Source RPM). In that vein the ``--use-wheel`` moniker would be kind of confusing to people.

Maybe that was a bad idea, I don't know.